lib/engine/game/g_1824/step/buy_sell_par_exchange_shares.rb

```diff
@@ -151,6 +151,14 @@ def allow_president_change?(corporation)
             reserved.none? { |s| s.percent == 20 }
           end
 
+          def action_is_shenanigan?(_entity, _other_entity, _action, _corporation, _corp_buying)
```

Comment:
```suggestion
          def action_is_shenanigan?(_entity, _other_entity, action, _corporation, _corp_buying)
```

Reply:
You're not testing any parameters passed to this method at the moment.

Reply:
I meant that you've got a mistake in the parameter names: the third parameter is called `_action`. The code in the method is testing `action`, which is not the parameter passed to the method.

Actually, since you are calling `super`, none of the parameters should start with an underscore. The declaration should be:
```suggestion
          def action_is_shenanigan?(entity, other_entity, action, corporation, corp_buying)
```